numpy/core/tests/test_multiarray.py

```diff
@@ -969,29 +969,6 @@ def test_sequence_long(self):
         assert_equal(np.array([long(4), 2**80, long(4)]).dtype, object)
         assert_equal(np.array([2**80, long(4)]).dtype, object)
 
-    def test_sequence_of_array_like(self):
-        class ArrayLike:
-            def __init__(self):
-                self.__array_interface__ = {
-                    "shape": (42,),
-                    "typestr": "<i1",
-                    "data": bytes(42)
-                }
-
-            # Make sure __array_*__ is used instead of Sequence methods.
```

Comment:
Nvm. Removed the test for now, because `__iter__` is now used again (tripping this assertion).